File: BubbleSort.rb
def BubbleSort(nums)
	swapped = true
	while swapped
		swapped = false
		(0..nums.size-2).each do |i|
			if nums[i] > nums[i+1]
				temp = nums[i+1]
				nums[i+1] = nums[i]
				nums[i] = temp
				swapped = true
			end
		end
	end
	nums
end

puts BubbleSort([9,8,7,6,5,4,3,2,1]).inspect

File: BubbleSort.js
function BubbleSort(nums) {
	var swapped = true;
	while (swapped) {
		swapped = false;
		for (var i = 0; i < nums.length-1; i++) {
			if (nums[i] > nums[i+1]) {
				var temp  = nums[i+1];
				nums[i+1] = nums[i];
				nums[i]   = temp;
				swapped = true;
			}
		}
	}
	return nums;
}

console.log( BubbleSort([9,8,5,4,2]));
console.log( BubbleSort([1,3,4,0,1]));